deposit/src/test/java/edu/unc/lib/deposit/fcrepo4/RegisterToLongleafJobTest.java

```diff
@@ -0,0 +1,240 @@
+
+        String output = FileUtils.readFileToString(new File(outputPath));
+
+        assertEquals("", output);
+    }
+
+    @Test
+    public void registerWorkWithMultipleFilesToLongleaf() throws Exception {
+        Bag workBag = addContainerObject(depBag, Cdr.Work);
+        Resource fileResc1 = addFileObject(workBag, FILE_CONTENT1, true);
+        fileResc1.addProperty(CdrDeposit.storageUri, FILE_SCHEME + LOC2_ID + "/original_file");
+        workBag.addProperty(Cdr.primaryObject, fileResc1);
+        Resource fileResc2 = addFileObject(workBag, FILE_CONTENT2, true);
+        fileResc2.addProperty(CdrDeposit.storageUri, FILE_SCHEME + LOC3_ID);
+
+        job.closeModel();
+
+        job.run();
+
+        String registrationArguments = "register -f " + LOC2_ID + "/original_file --checksums 'md5:" + MD5 + "'\n" +
+                "register -f " + LOC3_ID + "\n";
+        String output = FileUtils.readFileToString(new File(outputPath));
+
+        assertEquals(registrationArguments, output);
+    }
+
+    @Test
+    public void registerModsFilesToLongleaf() throws Exception {
+        Bag workBag = addContainerObject(depBag, Cdr.Work);
+        Resource fileResc = addFileObject(workBag, FILE_CONTENT1, true);
+        fileResc.addProperty(CdrDeposit.descriptiveStorageUri, FILE_SCHEME + LOC3_ID);
+        workBag.addProperty(Cdr.primaryObject, fileResc);
+
+        job.closeModel();
+
+        job.run();
+
+        String registrationArguments = "register -f " + LOC3_ID + "\n";
+        String output = FileUtils.readFileToString(new File(outputPath));
+
+        assertEquals(registrationArguments, output);
+    }
+
+    @Test
+    public void registerFitsExtractFilesToLongleaf() throws Exception {
+        Bag workBag = addContainerObject(depBag, Cdr.Work);
+        Resource fileResc = addFileObject(workBag, FILE_CONTENT1, true);
+        fileResc.addProperty(CdrDeposit.fitsStorageUri, FILE_SCHEME + LOC3_ID);
+        workBag.addProperty(Cdr.primaryObject, fileResc);
+
+        job.closeModel();
+
+        job.run();
+
+        String registrationArguments = "register -f " + LOC3_ID + "\n";
+        String output = FileUtils.readFileToString(new File(outputPath));
+
+        assertEquals(registrationArguments, output);
+    }
+
+    @Test
+    public void registerPremisLogFilesToLongleaf() throws Exception {
+        Bag workBag = addContainerObject(depBag, Cdr.Work);
+        Resource fileResc = addFileObject(workBag, FILE_CONTENT1, true);
+        fileResc.addProperty(CdrDeposit.premisStorageUri, FILE_SCHEME + LOC3_ID);
+        workBag.addProperty(Cdr.primaryObject, fileResc);
+
+        job.closeModel();
+
+        job.run();
+
+        String registrationArguments = "register -f " + LOC3_ID + "\n";
+        String output = FileUtils.readFileToString(new File(outputPath));
+
+        assertEquals(registrationArguments, output);
+    }
+
+    @Test
+    public void registerFilesWhichHaveNotBeenIngestedToLongleaf() throws Exception {
```

Comment:
Just to make sure i understand this test, its to verify that a file that is in the deposit but has no storage uri is not registered to longleaf?

Reply:
Yes, I think this case probably would never happen since anything that gets this far would have a storageUri value.  I can remove it if it seems unnecessary.

Reply:
No i think this is fine, just wanted to make sure i understood